primitives/tf_encrypted/primitives/sodium/cc/ops.cc

```diff
@@ -23,7 +42,22 @@ REGISTER_OP("SodiumEasyBoxSealDetached")
     .Input("sk_sender: uint8")
     .Output("ciphertext: uint8")
     .Output("mac: uint8")
-    .SetIsStateful();
+    .SetIsStateful()
+    .SetShapeFn([](shape_inference::InferenceContext* c){
+        shape_inference::ShapeHandle mac_shape = c->MakeShape({crypto_box_MACBYTES});
+        c->set_output(1, mac_shape);
```

Comment:
I would keep these in order, ie move logic for mac shape below logic for ciphertext shape

Reply:
sorry, pushing this change :D